include/jwt-cpp/jwt.h

```diff
@@ -2676,8 +2853,34 @@ namespace jwt {
 		 * \return *this to allow chaining
 		 */
 		verifier& issued_at_leeway(size_t leeway) {
-			return with_claim("iat", basic_claim_t(std::chrono::system_clock::from_time_t(leeway)));
+			claims["iat"] = verify_ops::date_after_claim<json_traits>{leeway};
+			return *this;
 		}
+
+		/**
+		 * Set an type to check for.
+		 * Check is casesensitive.
+		 * \param iss Issuer to check for.
+		 * \return *this to allow chaining
+		 */
+		verifier& with_type(typename json_traits::string_type type) {
+			std::transform(type.begin(), type.end(), type.begin(), [](char c) { return std::tolower(c); });
+			return with_claim("typ", [type](const verify_ops::verify_context<json_traits>& ctx, std::error_code& ec) {
+				if (!ctx.jwt.has_header_claim(ctx.claim_key)) {
+					ec = error::token_verification_error::missing_claim;
+					return;
+				}
+				auto c = ctx.jwt.get_header_claim(ctx.claim_key);
+				if (c.get_type() != json::type::string) {
+					ec = error::token_verification_error::claim_type_missmatch;
+					return;
+				}
+				auto val = c.as_string();
+				std::transform(val.begin(), val.end(), val.begin(), [](char c) { return std::tolower(c); });
```

Comment:
didn't know this was required https://datatracker.ietf.org/doc/html/rfc7519#section-5.1

Reply:
Neither did I, but reading the standard it appearently is.
It hasn't been an issue before though since we did not provide a way to explicitly check the type (since in 99.9% it won't matter), but somehow people seem to like doing this (theres now a check on the jwt.io library list for specifically this) so I dropped it in along the other changes.